src/main/java/com/aws/greengrass/util/platforms/WindowsPlatform.java

```diff
@@ -153,4 +158,40 @@ public ShellDecorator withShell(String shell) {
             throw new UnsupportedOperationException("changing shell is not supported");
         }
     }
+
+    @Inject
+    private Configuration config;
+
+//    public static final String NUCLEUS_DOMAIN_SOCKET_FILEPATH = "AWS_GG_NUCLEUS_DOMAIN_SOCKET_FILEPATH";
+//    public static final String NUCLEUS_DOMAIN_SOCKET_FILEPATH_FOR_COMPONENT =
+//            "AWS_GG_NUCLEUS_DOMAIN_SOCKET_FILEPATH_FOR_COMPONENT";
+
+
+    @Override
+    public String prepareDomainSocketFilepath() {
+        String ipcServerSocketAbsolutePath = "\\\\.\\pipe\\NucleusNamedPipe";
```

Comment:
this will need to be unique so that we don't conflict with multiple GGs on a single device. Maybe just add a UUID.